element-connector/src/main/java/org/eclipse/californium/elements/tcp/TcpClientConnector.java

```diff
@@ -59,10 +67,11 @@ public TcpClientConnector(int numberOfThreads, int connectTimeoutMillis, int idl
 		this.numberOfThreads = numberOfThreads;
 		this.connectionIdleTimeoutSeconds = idleTimeout;
 		this.connectTimeoutMillis = connectTimeoutMillis;
-		this.listenUri = URI.create(String.format("%s://127.0.0.1:0", getSupportedScheme()));
+		this.listenUri = URI.create(String.format("%s://0.0.0.0:0", getSupportedScheme()));
```

Comment:
>  then we can leave it as it is :-)

Let's use the provided address. If this is "localhost", OK, if it's "any", also OK.
But always use "127.0.0.1" is in my opinion missleading.